test/lib/autosetup_helpers_test.rb

```diff
@@ -50,6 +50,62 @@ class DummyClient < Yast::Client
     end
   end
 
+  describe "#suse_register" do
+    let(:profile_content) { { "general" => {} } }
+    let(:reg_module_available) { true }
+
+    before do
+      allow_any_instance_of(Y2Autoinstallation::AutosetupHelpers).to receive(
+        :registration_module_available?).and_return(reg_module_available)
+      allow(Yast::Profile).to receive(:current).and_return(profile_content)
+    end
+
+    context "yast2-register is not available" do
+      let(:reg_module_available) { false }
+      it "returns true" do
+        # no scc_auto call at all
+        expect(Yast::WFM).not_to receive(:CallFunction)
+        expect(client.suse_register).to eq(true)
+      end
+    end
+
+    context "yast2-register is available" do
+      let(:reg_module_available) { true }
+
+      context "suse_register tag is not defined in AY file" do
+        it "returns true" do
```

Comment:
I would add some more info here like *does not call any client and returns true*. What do you think? And the same for the rest.